samples/test-samples.cmake

```diff
@@ -50,7 +50,7 @@ if (BUILD_ENCLAVES)
     list(APPEND CRYPTO_LIB_LIST mbedtls)
   endif ()
 
-  if (UNIX)
+  if (UNIX AND CMAKE_SYSTEM MATCHES "Linux-5*")
```

Comment:
Add comments to explain why

Reply:
I believe the intention is to use CMAKE_SYSTEM to match whether the system OS is either Ubuntu 22.04 or Ubuntu 20.04 with `Linux-6*` and `Linux-5*` respectively. However, there could be two potential pitfalls:
1. This doesn't really help distinguish between different distributions of Linux e.g. RHEL vs Ubuntu
2. The version uses the kernel version by default so Ubuntu 20.04 with kernel 6* will be thought of as Ubuntu 22.04
These cases won't affect our CI and Ubuntu 20.04 VMs (with default marketplace image) on Azure, but it may affect others building our tests.

I recommend an alternative: check `/etc/os-release` which is typically available on Linux systems.